src/coreclr/jit/gentree.cpp

```diff
@@ -20493,8 +20522,14 @@ GenTree* Compiler::gtNewSimdBinOpNode(
                 if (varTypeIsLong(simdBaseType) || (simdBaseType == TYP_DOUBLE))
                 {
                     assert(varTypeIsSigned(simdBaseType));
-                    assert(compIsaSupportedDebugOnly(InstructionSet_AVX512F_VL));
-                    intrinsic = NI_AVX512F_VL_ShiftRightArithmetic;
+                    if (compOpportunisticallyDependsOn(InstructionSet_AVX10v1))
+                    {
+                        intrinsic = NI_AVX10v1_ShiftRightArithmetic;
+                    }
+                    else
+                    {
```

Comment:
Same consideration as above, we should preserve `assert(compIsaSupportedDebugOnly(InstructionSet_AVX512F_VL));` here in the else block